File: examples/m1n/assignment.ts
import util, { pipe } from "@std/util";
import math from "@std/math";

let val = 0;

const self = {};

function setVal(v) {
  return pipe(v).op(util["assign"], (_) => {
    val = _;
  }).end;
}
export default { ...self };

File: src/m1n/renderer/assignment.rs
use crate::{
    m1n::grammar::assingment_operation,
    parser_generator::render::{Render, RenderContext},
};

use super::Context;

impl Render<Context> for assingment_operation {
    fn render(&self, context: &mut RenderContext<Context>) {
        if let Some(value) = context.get_context().single_expression.clone() {
            context
                .str("util['assign'](")
                .render_boxed(&value)
                .str(", (_) => { ")
                .render_raw(&self.identifier)
                .str(" = _; })");
        } else {
            context
                .str("util['assign'], (_) => { ")
                .render_raw(&self.identifier)
                .str(" = _; }");
        }
    }
}
